src/python/zquantum/core/estimation/_estimation.py

```diff
@@ -215,62 +214,67 @@ def split_constant_estimation_tasks(
-            constant terms.
+        indices_to_not_measure: A list containing the indices of the EstimationTasks for
+            constant terms or with 0 shot.
     """
 
     estimation_tasks_to_measure = []
-    estimation_tasks_for_constants = []
+    estimation_tasks_not_measured = []
     indices_to_measure = []
-    indices_for_constants = []
+    indices_to_not_measure = []
     for i, task in enumerate(estimation_tasks):
-        if len(task.operator.terms) == 1 and () in task.operator.terms.keys():
-            indices_for_constants.append(i)
-            estimation_tasks_for_constants.append(task)
-        elif task.number_of_shots == 0:
-            raise RuntimeError(
-                "An EstimationTask requested 0 shot for a non-constant term. "
-                "It's unclear what to do with that."
-            )
+        if (
+            len(task.operator.terms) == 1 and () in task.operator.terms.keys()
+        ) or task.number_of_shots == 0:
+            indices_to_not_measure.append(i)
+            estimation_tasks_not_measured.append(task)
         else:
             indices_to_measure.append(i)
             estimation_tasks_to_measure.append(task)
 
     return (
         estimation_tasks_to_measure,
-        estimation_tasks_for_constants,
+        estimation_tasks_not_measured,
         indices_to_measure,
-        indices_for_constants,
+        indices_to_not_measure,
     )
 
 
-def evaluate_constant_estimation_tasks(
+def evaluate_non_measured_estimation_tasks(
     estimation_tasks: List[EstimationTask],
 ) -> List[ExpectationValues]:
-    """This function evaluates a list of EstimationTask over constant terms.
+    """This function evaluates a list of EstimationTask that are not
+    measured, and either contain only a constant term or require 0 shot.
+    Non-constant EstimationTask with 0 shot return 0.0 as their
+    ExpectationValue, with a precision of 0.0 as well.
 
     Args:
         estimation_tasks: The list of estimation tasks for which
-            Expectation Values are wanted, they must only contain constant terms.
+            Expectation Values are wanted.
 
     Returns:
-        expectation_values: the expectation values over constant terms,
+        expectation_values: the expectation values over non-measured terms,
             with their correlations and estimator_covariances.
     """
 
     expectation_values = []
     for task in estimation_tasks:
         if len(task.operator.terms) > 1 or () not in task.operator.terms.keys():
-            raise RuntimeError(
-                "evaluate_constant_estimation_tasks received an EstimationTask "
-                "that contained a non-constant term."
-            )
-        coefficient = task.operator.terms[()]
+            if task.number_of_shots > 0:
+                raise RuntimeError(
+                    "An EstimationTask required shots but was classified as a non-measured task"
```

Comment:
This line is too long to pass the style checks. Each line should be 88 characters or less.

Reply:
Fixed in 7271c6240983bb6beaef7c480e7deb773e8ad567